### PR TITLE
feat(shared-rank):[skp-181] Creates shared Rank component

### PR DESCRIPTION
**Pull request by Gardemarin**

### CODE REVIEW

File: libs/shared/rank/src/lib/shared-rank.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import { Rating } from '@material-ui/lab';
+
+import './shared-rank.scss';
+
+export interface SharedRankProps {
+  rank: number;
+}
+
+export const SharedRank = (props: SharedRankProps) => {
+  const [rank, setRank] = React.useState(props.rank);
```
Comment (doncevas) [inline libs/shared/rank/src/lib/shared-rank.tsx:12]: Don't use " React.useState(props.rank); " use  " useState(props.rank);"

File: libs/shared/rank/src/lib/shared-rank.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    <div className="rank">
+      <Rating
+        name="hover-feedback"
+        value={rank}
+        precision={0.5}
+        onChange={(event, newRank) => {
+          setRank(newRank);
+        }}
+      />
+      {rank !== null && (
```
Comment (doncevas) [inline libs/shared/rank/src/lib/shared-rank.tsx:24]: "rank !== null " use "!!rank && "

File: libs/shared/rank/src/lib/shared-rank.tsx (hunk 1 above)
Comment (doncevas) [inline libs/shared/rank/src/lib/shared-rank.tsx:12]: Don't use " React.useState(props.rank); " use " useState(props.rank);"  and useState should have default value, from props you can initialise only with useEffect

File: libs/shared/rank/src/lib/shared-rank.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import React, { useState, useEffect } from 'react';
+import { Box } from '@material-ui/core';
+import { Rating } from '@material-ui/lab';
+
+import './shared-rank.scss';
+
+export interface SharedRankProps {
+  value: number;
+  isRanked: boolean;
+  onUpdate: (data) => void;
```
Comment (MariuzM) [inline libs/shared/rank/src/lib/shared-rank.tsx:10]: Please add type 
image

File: libs/shared/rank/src/lib/shared-rank.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  }, [props]);
+
+  return (
+    <div className="rank">
+      <Rating
+        disabled={rank.isRanked}
+        name="hover-feedback"
+        value={rank.value}
+        precision={0.5}
+        onChange={(event: EventTarget, newValue: number) => {
```
Comment (MariuzM) [inline libs/shared/rank/src/lib/shared-rank.tsx:30]: Please fix this warning 
image

File: libs/shared/rank/src/lib/shared-rank.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+          if (!rank.isRanked) {
+            let newRank = { isRanked: true, value: newValue };
```
Comment (MariuzM) [inline libs/shared/rank/src/lib/shared-rank.tsx:32]: I would recommend to write in `const` first and if the var is going to change, change to `let` 
image

**Pull request merged**